README.adoc

```diff
@@ -1 +1,82 @@
 = Shell Function Invoker image:https://ci.projectriff.io/api/v1/teams/main/pipelines/riff/jobs/build-shell-function-invoker-container/badge[Shell Invoker Status, link=https://ci.projectriff.io/teams/main/pipelines/riff/build-shell-function-invoker-container/builds/latest]
+
+== Purpose
+The *shell function invoker* accepts gRPC requests, invokes a command for each request in the input stream,
+and send the command's output to the stream of gRPC responses.
+
+== Development
+=== Prerequisites
+The following tools are required to build this project:
+
+- `make`
+- Docker
+- https://github.com/Masterminds/glide#install[Glide] for dependency management
+
+If you intend to re-generate mocks for testing, install:
+
+- https://github.com/vektra/mockery#installation[Mockery]
+
+If you would like to run tests using the `ginkgo` command, install:
+
+- http://onsi.github.io/ginkgo/[Ginkgo]
+
+If you need to re-compile the protobuf protocol, install:
+
+- Google's https://github.com/google/protobuf[protocol compiler]
+
+=== Get the source
+[source, bash]
+----
+cd $(go env GOPATH)   #defaults to ~/go
+git clone -o upstream https://github.com/projectriff/shell-function-invoker src/github.com/projectriff/shell-function-invoker
+----
+
+=== Building
+* To build locally (this will produce a binary named `shell-function-invoker` on _your_ machine):
++
+[source, bash]
+----
+make build
+----
+
+* To build for docker (so that it can be deployed along with a command-based function):
```

Comment:
That sentence is not very clear. See my comment in the "purpose" section

Reply:
Clarified.